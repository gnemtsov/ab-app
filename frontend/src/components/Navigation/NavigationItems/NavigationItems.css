.NavigationItems {
    margin: 2rem 0 0 0;
    padding: 0;
    list-style: none;
    display: flex;
    flex-flow: column;
    align-items: center;
    height: 100%;
}

@media (min-width: 500px) {
    .NavigationItems {
        margin: 0;
        flex-flow: row;
    }
}