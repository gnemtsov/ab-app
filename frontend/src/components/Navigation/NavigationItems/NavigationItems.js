import React from 'react';

import classes from './NavigationItems.css';
import NavigationItem from './NavigationItem/NavigationItem';

const navigationItems = ( props ) => (
    <ul className={classes.NavigationItems}>
        <NavigationItem link="/" exact>Departments</NavigationItem>
        {!props.isAuthenticated
            ? <NavigationItem link="/auth">Authenticate</NavigationItem>
            : <NavigationItem link="/logout">Logout</NavigationItem>}
    </ul>
);

export default navigationItems;