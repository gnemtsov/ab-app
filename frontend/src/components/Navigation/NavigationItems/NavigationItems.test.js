import React from 'react';

import { configure, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter } from "react-router-dom";

import NavigationItems from './NavigationItems';

configure({ adapter: new Adapter() });

describe('NavigationItems', () => {
	it('render with authentication', () => {
		const wrapper = render(
			<BrowserRouter>
				<NavigationItems link="/" isAuthenticated>Departments</NavigationItems>
			</BrowserRouter>
		);
		
		expect(wrapper).toMatchSnapshot();
	});
	
	it('render without authentication', () => {
		const wrapper = render(
			<BrowserRouter>
				<NavigationItems link="/">Departments</NavigationItems>
			</BrowserRouter>
		);
		
		expect(wrapper).toMatchSnapshot();
	});
});
