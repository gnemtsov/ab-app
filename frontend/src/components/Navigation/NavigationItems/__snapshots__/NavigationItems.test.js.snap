// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItems render with authentication 1`] = `
<ul
  class="NavigationItems"
>
  <li
    class="NavigationItem"
  >
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Departments
    </a>
  </li>
  <li
    class="NavigationItem"
  >
    <a
      href="/add"
    >
      Add
    </a>
  </li>
  <li
    class="NavigationItem"
  >
    <a
      href="/logout"
    >
      Logout
    </a>
  </li>
</ul>
`;

exports[`NavigationItems render without authentication 1`] = `
<ul
  class="NavigationItems"
>
  <li
    class="NavigationItem"
  >
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Departments
    </a>
  </li>
  <li
    class="NavigationItem"
  >
    <a
      href="/auth"
    >
      Authenticate
    </a>
  </li>
</ul>
`;
